01-spring-boot/README.md

```diff
@@ -142,4 +142,41 @@ CONTAINER ID        IMAGE               COMMAND                  CREATED
 46c5781e269f        apiworkshop:v1      "sh -c 'java $JAVA_O…"   About a minute ago   Up About a minute   0.0.0.0:8081->8080/tcp   peaceful_stonebraker
 ```
 
-If you need to restart/stop the container run `docker kill <container ID>`.
\ No newline at end of file
+If you need to restart/stop the container run `docker kill <container ID>`.
+
+### Step 6 (Optional) - Building with Docker
+
+One limitation with our above solution is that we have to remember to build both with gradle and then with docker. 
+This is both easy to forget and makes creating a build pipeline that bit more complex.
+
+There is a solution, and that is that we modify the `Dockerfile` to both build the java code and create the runnable image. 
+
+```
+FROM gradle:jdk12 as builder
+
+COPY --chown=gradle:gradle . /home/gradle/src
+WORKDIR /home/gradle/src
+RUN gradle build
+```
+If we add the above to the beggining of our existing Dockerfile it will have the effect of creating a _Multistage build_ that is we will use the result of one stage in the stage that packages up the jar for running. 
```

Comment:
Maybe tag in the docker file best practices as well. I think that is a nice explanation of why this is a good end state